io/pci/pci_hotplug.py

```diff
@@ -128,16 +129,71 @@ def is_recovered():
+                disks = pci.get_disks_in_pci_address(pci_addr)
+                err_disks = []
+                for disk in disks:
+                    curr_path = disk.split("/")[-1]
+                    self.log.info("curr_path=%s" % curr_path)
+                    if not wait.wait_for(is_path_online, timeout=10):
+                        self.log.info("%s failed to recover after add" % disk)
+                        err_disks.append(disk)
+                    else:
+                        self.log.info("%s Recovered successfully" % disk)
+
+            if err_disks:
+                self.log.info("few paths failed to recover : %s" % err_disks)
+                return False
+            else:
+                self.log.info("All Paths recovered successfully")
                 return True
-            return False
 
-        if not wait.wait_for(is_added, timeout=10):
-            return False
-        # Waiting for 10s per end device, for recovery.
-        return wait.wait_for(is_recovered,
-                             self.end_devices[pci_addr] * 10)
+        def net_recovery_check():
+            """
+            calls different recovery functions based on adapter type
+            """
+            local = LocalHost()
+            iface = []
+            iface = pci.get_interfaces_in_pci_address(pci_addr, 'net')
+            self.log.info("interface_value=%s, %s" % (iface, iface[0]))
+            networkinterface = NetworkInterface(iface[0], local)
+            self.log.info("checking link is up or not")
+            if wait.wait_for(networkinterface.is_link_up(), timeout=120):
+                self.log.info("Link is UP after adapter added back")
+                self.log.info("now check adapter is pingable or not")
+                if networkinterface.ping_check(self.peer_ip, count=5) is None:
+                    self.log.info("inteface is up and pinging")
+                    return True
+                else:
+                    self.log.info("interface up but not pinging")
+                    return False
+            else:
+                self.log.info("interface is not up after hotplug")
+                return False
+
+        if wait.wait_for(is_added, timeout=10):
+            if pci.get_pci_class_name(pci_addr) is 'net':
+                if wait.wait_for(net_recovery_check, timeout=10):
+                    self.log.info("Adapter added and recovered successfully")
+                    return True
+                else:
+                    self.log.info("Adapter added back but but not pinging")
+                    return False
+            else:
+                if wait.wait_for(is_recovered, timeout=10):
+                    self.log.info("Adapter added and recovered successfully")
+                    return True
+                else:
+                    self.log.info("Adapter added back but disks not recovered")
+                    return False
+        return False
```

Comment:
so many else and log info,, you can remote and log only if fails/not reocvered

Reply:
@abdhaleegit 
since this started with premature level. so all these massages are added to get the failing part for quick resolution.
will send the final patch taking care of cosmetic changes.
hope all technical things are  ok.